lib/legacy/class.html

```diff
@@ -96,15 +95,26 @@
     // If lifecycle is called (super then me), order is
     // (1) C.created, (2) A.created, (3) B.created, (4) element.created
     // (again same as 1.x)
-    function _mixinBehaviors(behaviors, klass) {
-      for (let i=0; i<behaviors.length; i++) {
-        let b = behaviors[i];
-        if (b) {
-          klass = Array.isArray(b) ? _mixinBehaviors(b, klass) :
-            GenerateClassFromInfo(b, klass);
+    function copyBehaviorProperties(behaviors, klass) {
+      const meta = {};
+      const superMeta = klass.prototype.__behaviorMetaProps;
+      if (behaviors) {
+        klass.prototype.__behaviorMetaProps = meta;
+        for (let i=0; i<behaviors.length; i++) {
+          copyProperties(behaviors[i], klass.prototype);
+          memoizeBehaviorMetaProps(meta, behaviors[i], superMeta);
+        }
+      }
+      klass.prototype.__behaviorMetaProps = meta;
+    }
+
+    function memoizeBehaviorMetaProps(meta, behavior, superMeta) {
```

Comment:
There's a bug here where the full flattened list of lifecycle callbacks is called once per call to `GenerateClassFromInfo`, which happens each time `mixinBehaviors` is called.

Instead, store the list of callbacks in the closure for one set of behaviors associated with each call to `GenerateClassFromInfo`.

Reply:
Fixed and test added.